std/socket.d

```diff
@@ -1983,6 +2000,8 @@ static if (is(sockaddr_un))
             sun.sun_family = AddressFamily.UNIX;
             sun.sun_path.ptr[0 .. path.length] = (cast(byte[]) path)[];
             sun.sun_path.ptr[path.length] = 0;
+            _nameLen = cast(socklen_t) (sockaddr_un.init.sun_path.offsetof +
+                path.length + (sun.sun_path.ptr[0] ? 1 : 0));
```

Comment:
Alright, so I understand that unlike regular UNIX sockets, abstract sockets' paths do not need to be null-terminated (and in fact doing so would change the address).

Please add a short comment describing the length calculation done here, as in your message above. A DDoc comment on the constructor mentioning how to create an abstract UNIX socket address would also be nice.